all_repos/push/bitbucket_server_pull_request.py

```diff
@@ -29,9 +30,14 @@ def make_pull_request(
         settings: Settings,
         branch_name: str,
 ) -> bitbucket_server_api.Response:
+    if settings.bearer_auth:
+        auth_header = {'Authorization': f'Bearer {settings.app_password}'}
+    else:
+        auth_header = {'Authorization': f'Basic {settings.auth}'}
```

Comment:
this to me feels like it should be a separate `source`.  `app_password` isn't actually the app password but a bearer token

Reply:
As per the documentation https://confluence.atlassian.com/bitbucketserver/http-access-tokens-939515499.html this password can be used for both `Basic` and `Bearer` auth. I am not sure why we need two separate source.

Reply:
they are fundamentally different auth models -- a bearer token does not have a username.  they should be two separate sources rather than a boolean trap

Reply:
@asottile Alright, I created separate source and push for this.